### DISCUSSION TITLE
Failing upon connecting to web server

### DISCUSSION THREAD

**Discussion by Mahallon** [tobozo/WUD-Ducky#22] [Q&A]
I got the April brothers Device and I was able to flash the newest branch 2.0.3-RC1 from this repository onto the device. 
When I plug in the device I get a solid red LED, The WUD-01 AP is created and I can connect to it with the default password 12345678.
After connecting to the network If I navigate to 192.168.4.1 in browser and I get the default connection screen where it askes if I understand. As soon as I click the link saying that I understand the red LED turns off and I hear the USB device disconnected sound from my computer. After the network comes back, there are no files in the file list, the info page seems to be just full of default information and the logs aren't helpful (to me). It does that that LittleFS failed (but I see the readme file in file explorer when the device is plugged in) and I have tried with and without SD card inserted. At one point I was able to start DrawSpaceInvaders but it disconnected half way through drawing it.  I was able to turn on and off the NUMLK and CAPSLK lights from the info page as well but that was only when connecting to the network from the computer that the device was plugged into and not from my mobile phone connected to the device network. Same thing with the touchpad type interface working on the computer but not my phone. 

What other steps do I need to do to debug this? I was thinking about trying an earlier branch but it seems like I have lost the ability to flash to the device. I used the button and magnet to get it to allow me to flash your software but now it will not. I thought I saw something about OTA but have not been able to get that working either. Please Help.

![InfoBottom](https://github.com/tobozo/WUD-Ducky/assets/13002995/6731ce47-4bcc-4de4-976e-470d23b16b67)
![InfoTop](https://github.com/tobozo/WUD-Ducky/assets/13002995/306951a2-7df1-44c6-8db0-7cf01cdbda2a)
![WUDLogs](https://github.com/tobozo/WUD-Ducky/assets/13002995/e576fd20-e148-4a9d-a828-df843addc50a)

**Comment by tobozo** [comment]
hi,

don't forget to upload LittleFS after flashing :-)

![image](https://github.com/tobozo/WUD-Ducky/assets/1893754/aed1366d-2f10-495a-a406-3ac05d50bd0c)

https://randomnerdtutorials.com/esp32-littlefs-arduino-ide/

**Comment by Mahallon** [comment]
Thank you for your quick reply @tobozo, I need to download Arduino 1.8.19 and set it up since the LittleFS doesn't work with 2.x.x and I'll let you know if that solves the problem

**Comment by Mahallon** [comment]
I downloaded Arduino 1.8.19 as well as installing LittleFS. As per the tutorial I now have the ESP32 Sketch Data Upload option under tools but when I try and upload it always fails. I went back and read the tutorial you sent me again and saw that the serial monitor needed to be closed so I verified that and tried again but it still failed. The tutorial also mentioned that some boards need you to press the onboard BOOT button for 2 seconds so I tried pressing the onboard button a few times and it also still failed, I tried on several different COM ports and over IP as well all of which failed, getting desperate I tried with and without SD card plugged in, I tried different baud rates. I factory reset the board and tried to flash the LittleFS files before flashing 2.0.3-RC1 back to the board and that also failed. Any tips? I'm pulling my hair out at the point... what I want to do with this is a fraction of this repo's full potential I don't even want to use LittleFS I want a WiFi controlled mouse and keyboard. Should I use an earlier branch of this project or is there a different one you would recommend?

**Comment by tobozo** [comment]
you can use the same method for uploading LittleFS and flashing the firmware (e.g. put a magnet on the ESP32-WUD before powering up, or do the RST+BOOT buttons push), but the partition size must match reality otherwise LittleFS upload will fail

https://wiki.aprbrother.com/en/wud.html#features <<< flash size is 4MB

**Comment by Mahallon** [comment]
I thought I read somewhere on the 2.0.3 documentation that the magnet thing wasn:t necessary on that version. is that something that I have to set up? Just for clarification, I put the magnet up against the device before powering it up? Also to take advantage of your awesome skill to quickly reply, what is the RTS plus boot button? Is that the button that's on the device and it has two functions?

**Reply by tobozo** [reply]
partition scheme and flash size are in the tools menu, flash size must match your hardware, and partition scheme must match the flash size

![image](https://github.com/tobozo/WUD-Ducky/assets/1893754/b85a2958-4235-4045-b830-87282f17ea97)

**Reply by Mahallon** [reply]
Thank you I will try that

**Comment by Mahallon** [comment]
The Flash size is the correct size. Attached is the Tools view along with the latest failure

putting the magnet up against the device before powering on leads to the device showing up on COM10 Port and a timeout failure for the LittleFS upload 

Pressing the button on the device leads to the device showing up on COM5 Port and it looks like the whole thing is going to work, the upload gets to 100% but it fails at the very end because it cant reset it?
![littleFSUploadFail3](https://github.com/tobozo/WUD-Ducky/assets/13002995/bc5ed412-4040-4051-ab43-9c168cf2183a)

**Reply by tobozo** [reply]
if the upload and hash data verification succeeded the last error message may be a false negative (only the reset failed)

**Reply by Mahallon** [reply]
I am no longer seeing the LIttleFS failed messages in the logs so I think you are right that its a false negative. 

![AfterLIttleFSInstall2](https://github.com/tobozo/WUD-Ducky/assets/13002995/8094758a-3845-4b6d-bba4-806f722926bd)

Unfortunately none of the other behaviors improved, Below is how long It was able to draw space Invaders before it seemingly disconnected itself

![SpaceInvader](https://github.com/tobozo/WUD-Ducky/assets/13002995/25fe77a7-b465-4675-952b-3e03deb5c96e)

The info page is still full of nothing as well

**Comment by Mahallon** [comment]
As a Sanity check, I flashed the 2.0.2 branch onto the device and it is working properly now.

**Reply by Mahallon** [reply]
Correction: info page is working as intended with real information, LittleFS file list is showing up in WebUI: touchpad like interface and NUM/Caps/Scroll lock is now working on phone interface but  DrawSpaceInvaders still causes the device to disconnect. At this point I wonder if it is Windows detecting non-human movement from the HID device and quarantining it automatically? Just conjecture

**Reply by tobozo** [reply]
or it's just a bug in the code, bad priority or too small stack size in the draw task maybe, hard to say as it may vary depending on the espressif core version
